plugins/dialogui/plugin.js

```diff
@@ -1011,14 +1011,22 @@ CKEDITOR.plugins.add( 'dialogui', {
 			 * Selects all the text in the text input.
 			 */
 			select: function() {
-				var me = this.selectParentTab();
+				var me = this.selectParentTab(),
+					editor = this._.dialog._.editor;
 
 				// GECKO BUG: setTimeout() is needed to workaround invisible selections.
 				setTimeout( function() {
-					var e = me.getInputElement();
+					var e = me.getInputElement(),
+						ranges = editor.getSelection().getRanges();
+
 					if ( e ) {
 						e.$.focus();
 						e.$.select();
+
+						// #1686 Edge occasionally loses editor selection. Restore it when it happens.
```

Comment:
It seems like a more general fix, which IMO should be moved to the separate issue and some tests should be added to it.

Reply:
I can't reproduce this problem manually. It occurs only in this case:

```
range.setStart( paragraph, 1 );
range.setEnd( paragraph, 3 );
range.select();
editor.execCommand( 'link' );
```

And focus is lost above in line 1023 by native method.

Reply:
Yet it's valid bug, connected with API. It should have dedicated ticket with tests etc.

Reply:
I spend some time testing it, and I can't reproduce this issue on Edge 17, hence I'm dropping that fix.